src/dice.class.php

```diff
@@ -129,7 +129,7 @@ function keygen() {
 		$pass = "";
 		for($length = 1; $length < 24; $length++) {
```

Comment:
I didn't notice the index starting from one before, so that it's generating a 23-character password rather than a 24-character password.  No big deal.

On a whim, I checked the _key.dat_ files for both staging and production.  While production contains a 23-character password (that was generated 10 years ago), staging has a 22-character password (that was generated last month).  I'm trying to reverse engineer how that could have happened (again, it's NBD, but I'm curious).

Is it possible that at the time staging generated its password, it hit the 1/124 chance of indexing the string out of bounds, and that skipped a character?

Reply:
@ssoloff Yup, seems like it ^^
I changed the code so it would exactly be out of bound one time (because I already had deleted the key.dat file), and that resulted in a 22 chars long string.